homeassistant/components/hassio/__init__.py

```diff
@@ -455,4 +477,132 @@ async def async_handle_core_service(call):
     # Init add-on ingress panels
     await async_setup_addon_panel(hass, hassio)
 
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(DOMAIN, context={"source": "system"})
+    )
+
     return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    dev_reg = await async_get_registry(hass)
+    coordinator = HassioAddonsDataUpdateCoordinator(hass, config_entry, dev_reg)
+    hass.data[ADDONS_COORDINATOR] = coordinator
+    await coordinator.async_refresh()
+
+    def noop_listener():
+        """Noop listener to force coordinator to refresh."""
+        return
+
+    # add a noop listener to force coordinator to refresh since entities are disabled by default
+    hass.data[ADDONS_COORDINATOR_UNSUB_LISTENER] = coordinator.async_add_listener(
+        noop_listener
+    )
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+        )
+
+    return True
+
+
+async def async_unload_entry(
+    hass: HomeAssistantType, config_entry: ConfigEntry
+) -> bool:
+    """Unload a config entry."""
+    # Unsubscribe from coordinator listener
+    if ADDONS_COORDINATOR_UNSUB_LISTENER in hass.data:
+        hass.data[ADDONS_COORDINATOR_UNSUB_LISTENER]()
+        hass.data.pop(ADDONS_COORDINATOR_UNSUB_LISTENER)
+
+    # Pop add-on data
+    hass.data.pop(ADDONS_COORDINATOR, None)
+
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, platform)
+                for platform in PLATFORMS
+            ]
+        )
+    )
+
+    return unload_ok
+
+
+@callback
+def register_addons_in_dev_reg(
+    entry_id: str, dev_reg: DeviceRegistry, addons: List[Dict[str, Any]]
+) -> None:
+    """Register addons in the device registry."""
+    for addon in addons:
+        dev_reg.async_get_or_create(
+            config_entry_id=entry_id,
+            identifiers={(DOMAIN, addon[ATTR_SLUG])},
+            manufacturer=addon.get(ATTR_REPOSITORY) or addon.get(ATTR_URL) or "unknown",
+            model="Hass.io Add-On",
```

Comment:
We don't use the term Hass.io anymore in user-facing text.

```suggestion
            model="Home Assistant Add-on",
```